charts/vald/templates/rebalancer/jobconfigmap.yaml

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+{{- if .Values.rebalancer.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Values.rebalancer.rebalancer.rebalance_job_configmap_name }}
+  labels:
+    app.kubernetes.io/name: {{ include "vald.name" . }}
+    helm.sh/chart: {{ include "vald.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.Version }}
+    app.kubernetes.io/component: rebalancer
+data:
+  {{ .Values.rebalancer.rebalancer.rebalance_job_template_key }}: |
+    ---
+    apiVersion: batch/v1
+    kind: Job
+    metadata:
+      name: {{ .Values.rebalancer.rebalancer.rebalance_job_name }}
+      namespace: vald
+      labels:
+        app: {{ .Values.rebalancer.rebalancer.rebalance_job_name }}
+    spec:
+      failedJobsHistoryLimit: 2
+      successfulJobsHistoryLimit: 2
+      template:
+        spec:
+          containers:
+            - name: vald-rebalancer-job
+              image: {{ .Values.rebalancer.rebalancer.rebalance_job_image }}
+              imagePullPolicy: Always
+              env:
+                - name: AWS_ACCESS_KEY
+                  valueFrom:
+                    secretKeyRef:
+                      name: aws-secret
+                      key: access-key
+                - name: AWS_SECRET_ACCESS_KEY
+                  valueFrom:
+                    secretKeyRef:
+                      name: aws-secret
+                      key: secret-access-key
```

Comment:
sorry maybe I am not fully understand what you mean...
do you mean we should define another variable in values.yaml to set these env?

Reply:
thank you.
I fixed: https://github.com/vdaas/vald/pull/1187/commits/7add40b9790232a52b1213dce7d5088be65d4cb1

Reply:
In your commit, `RATE` and `TARGET_AGENT_NAME` will be passed from values.yaml. Is it okay? :thinking:

Reply:
thank you. :bow:
I think it is ok :bow:
Because `RATE` and `TARGET_AGENT_NAME` are dynamically set by the controller. 🤔 
cc: @kevindiu @vankichi

Reply:
I think it's not okay because the users should take care about these environmental variables.

There're hard coded variable names here, it means that the environmental variables are referred here.
https://github.com/vdaas/vald/blob/d436af744ce093c2c629aa5a4a82821c50c645ff/charts/vald/templates/rebalancer/jobconfigmap.yaml#L102-L103
If the users can edit the definitions of `RATE` and `TARGET_AGENT_NAME` through values.yaml, the consistency between the definitions and the references may be easily broken.

It's better to write these environmental variables in the template, like sidecar deployment:
https://github.com/vdaas/vald/blob/6b9bc98ba6212e3e25cabe4c39b27a6c5b6880c3/charts/vald/templates/agent/deployment.yaml#L145-L147

Reply:
> I think it's not okay because the users should take care about these environmental variables.
There're hard coded variable names here, it means that the environmental variables are referred here.

thank you for feedback.
I thought about it again, I made a mistake.
Adding environment variables dynamically makes them unmanageable and complicated.
sorry ...
I will fix it soon. thank you :bow:

Reply:
@vankichi @kevindiu 
what do you think?

Reply:
@rinx 
thank you.
I understand what you are saying. I misread the code. I'll fix it later. :bow:

Reply:
thank you for feedback.
I fixed: https://github.com/vdaas/vald/pull/1187/commits/0ee5fe1790b3ce329864f3f180f15c5794d10b01